mediorum/cidutil/cidutil.go

```diff
@@ -0,0 +1,73 @@
+package cidutil
+
+import (
+	"fmt"
+	"io"
+	"mime/multipart"
+	"path/filepath"
+
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multihash"
+)
+
+func ComputeFileHeaderCID(fh *multipart.FileHeader) (string, error) {
+	f, err := fh.Open()
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	return ComputeFileCID(f)
+}
+
+func ComputeFileCID(f io.ReadSeeker) (string, error) {
+	defer f.Seek(0, 0)
+	builder := cid.V1Builder{}
+	hash, err := multihash.SumStream(f, multihash.SHA2_256, -1)
+	if err != nil {
+		return "", err
+	}
+	cid, err := builder.Sum(hash)
+	if err != nil {
+		return "", err
+	}
+	return cid.String(), nil
+}
+
+// note: any Qm CID will be invalid because its hash won't match the contents
+func ValidateCID(expectedCID string, f io.ReadSeeker) error {
+	computed, err := ComputeFileCID(f)
+	if err != nil {
+		return err
+	}
+	if computed != expectedCID {
+		return fmt.Errorf("expected cid: %s but contents hashed to %s", expectedCID, computed)
+	}
+	return nil
+}
```

Comment:
these are unchanged. just moved to a util to avoid circular import when running the migration